### DISCUSSION TITLE
Need help regarding Expo ImageManipulator deprecated Method, or please suggest me a replaced method.

### DISCUSSION THREAD

**Discussion by majidmousav** [expo/expo#33830] [Archived: Questions]
Hello Everybody!
I need to do image Crop & Resize in my app using ImageManipulator library, but it seems my necessarily method has been deprecated. the official page suggested to use a hook or another method, but none of them work for me. Could you please suggest how to solve the issue? or if you have any recommendation(Or if you have any example which already work with new methods), would be really appreciate it.
![Capturasde](https://github.com/user-attachments/assets/d760ca0d-fb95-45de-a221-a3e3f5be1b1c)

**How should I modify below code?**
@brentvatne 
@amandeepmittal 

import React, { useState } from "react";
import { View, Button, Image, StyleSheet, Alert } from "react-native";
import * as ImagePicker from "expo-image-picker";
import { manipulateAsync, SaveFormat } from "expo-image-manipulator";

const ResizeWithManipulator = () => {
  const [imageUri, setImageUri] = useState(null);
  const [manipulatedUri, setManipulatedUri] = useState(null);

  const handleSelectImage = async () => {
    try {
      const result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.Images,
        allowsEditing: false,
      });

      if (!result.canceled && result.assets.length > 0) {
        setImageUri(result.assets[0].uri);
      }
    } catch (error) {
      console.error("Error selecting image:", error);
      Alert.alert("Error", "Failed to select image.");
    }
  };

  const handleManipulateImage = async () => {
    if (!imageUri) {
      Alert.alert("Error", "No image selected.");
      return;
    }

    try {
      const manipResult = await manipulateAsync(
        imageUri,
        [
          { crop: { originX: 50, originY: 50, width: 200, height: 200 } },
          { resize: { width: 300, height: 300 } },
        ],
        { compress: 0.8, format: SaveFormat.JPEG }
      );
      setManipulatedUri(manipResult.uri);
    } catch (error) {
      console.error("Error manipulating image:", error);
      Alert.alert("Error", "Failed to manipulate image.");
    }
  };

  return (
    <View style={styles.container}>
      <Button title="Select Image" onPress={handleSelectImage} />
      {imageUri && <Image source={{ uri: imageUri }} style={styles.imagePreview} />}
      <Button title="Manipulate Image" onPress={handleManipulateImage} disabled={!imageUri} />
      {manipulatedUri && (
        <Image source={{ uri: manipulatedUri }} style={styles.imagePreview} />
      )}
    </View>
  );
};

export default ResizeWithManipulator;

**Comment by GodGardensGuns** [comment]
I ran into a similar issue. The `saveOptions` methods don't seem to be available with `ImageManipulator.manipulate` yet. Namely `base64` and `compress`. I had to resort to using `manipulateAsync` for now.

**Comment by codegsaini** [comment]
@GodGardensGuns We can apply `SaveOptions` with the help of `saveAsync(options: SaveOptions)` function using `ImageManipulatorContext` in `expo-image-manipulator` version `~13.1.7` as of today.

<https://docs.expo.dev/versions/latest/sdk/imagemanipulator/#renderasync>

We have to get `ImageRef` provided by `renderAsync()` function which will return `Promise<ImageRef>`.

```js
const imageRef = await ImageManipulator.manipulate(uri).renderAsync();
```

Now on `ImageRef` we can apply `SaveOptions` with the help of `saveAsync(options: SaveOptions)` function.

```js
const imageResult = await imageRef.saveAsync(options: SaveOptions);
```

@majidmousav Your manipulation code can be replaced by below implementation.

```js
const imageRef = await ImageManipulator.manipulate(imageUri)
.crop({
    originX: 50,
    originY: 50,
    width: 200,
    height: 200,
})
.resize({
    width: 300,
    height: 300,
})
.renderAsync();
const manipResult = await imageRef.saveAsync({
    compress: 0.8,
    format: ImageManipulator.SaveFormat.JPEG,
});
setManipulatedUri(manipResult.uri);
```

Hope it helps.
